### PR TITLE
Update ZUGFeRD XML generation

### PR DESCRIPTION
**Pull request by cmcproject**
- Fix XML generation issue
	- "xml" and "rsm" elements were generated on the same line; add "rsm" element on the next line
- Add docstring for serialize method
- Update utils and tests
	- remove unused method
	- move tests related method

### CODE REVIEW

**Review by raphaelm — APPROVED** [review]

**Comment by cmcproject** [comment]
@raphaelm could you trigger a new release?

**Comment by raphaelm** [comment]
This PR does not change any relevant behaviour of the library, if I'm not mistaken, does it?

**Comment by cmcproject** [comment]
It only fixes the first line of the XML by addling the rsm element on a new line.
image

**Comment by raphaelm** [comment]
I don't see that this new line is required in the spec. Does it break anything?

**Comment by cmcproject** [comment]
Here is an official example of a ZUGFeRD PDF.
[BASIC-WL_Einfach.pdf](https://github.com/pretix/python-drafthorse/files/11771367/BASIC-WL_Einfach.pdf)

image

File: drafthorse/models/elements.py [hunk 1]
```diff
@@ -67,10 +67,14 @@ def append_to(self, node):
             node.append(el)
 
     def serialize(self, schema="FACTUR-X_BASIC"):
-        xml = b'<?xml version="1.0" encoding="UTF-8"?>' + ET.tostring(
```
Comment (cmcproject) [inline drafthorse/models/elements.py:70]: This concatenation issue was fixed by adding the "utf-8" parameter to tostring method.

File: drafthorse/models/elements.py [hunk 2]
```diff
@@ -67,10 +67,14 @@ def append_to(self, node):
...
-            self.to_etree(), "utf-8"
-        )
-        return validate_xml(xmlout=xml, schema=schema)
+        """
+        Create XML from ZUGFeRD data model
+        :param schema: XML schema
+        :return: ZUGFeRD XML
+        """
+        xml = ET.tostring(self.to_etree(), "utf-8")
```
Comment (cmcproject) [inline drafthorse/models/elements.py:75]: See here.

**Comment by raphaelm** [comment]
Yeah, I see the change. I don't yet get why it matters? It seems legal XML to me either way. Did it break in any tools?

**Comment by cmcproject** [comment]
> Yeah, I see the change. I don't yet get why it matters? It seems legal XML to me either way. Did it break in any tools?

It is an improvement on how the XML is being generated. It is more related to the readability then to the functional aspect.

**Comment by raphaelm** [comment]
Okay. I think it's a very cosmetic change that does not matter much. It's nice to have but I don't see how it needs a release.

**Pull request merged**